pandas/tests/frame/test_mutate_columns.py

```diff
@@ -165,6 +165,23 @@ def test_delitem(self):
         del self.frame['A']
         self.assertNotIn('A', self.frame)
 
+    def test_delitem_multiindex(self):
+        midx = MultiIndex.from_product([['A', 'B'], [1, 2]])
+        df = DataFrame(np.random.randn(4, 4), columns=midx)
+        assert len(df.columns) == 4
+
+        result = df['A']
+        assert isinstance(result, DataFrame)
+        del df['A']
+
+        assert len(df.columns) == 2
+
+        with pytest.raises(KeyError):
+            del df[('A',)]
```

Comment:
Is it the purpose that this also raises if the `A` key is still in `df`. If so, I would put this before `del df['A']`. Otherwise we also need to test that this works on the initial `df`